app/models/concerns/model/subtransactable.rb

```diff
@@ -13,8 +13,26 @@ module Model::Subtransactable
 		has_one :supporter, through: :trx
 		has_one :nonprofit, through: :trx
 
-		has_many :subtransaction_payments, through: :subtransaction
+		has_many :subtransaction_payments, through: :subtransaction  do
+			def gross_amount
+				map{|i| i.gross_amount}.sum
+			end
+	
+			def net_amount
+				map{|i| i.net_amount}.sum
+			end
+	
+			def fee_total
+				map{|i| i.fee_total}.sum
+			end
+	
+		end
 
 		delegate :currency, to: :nonprofit
+
+		# Handle a completed refund from a legacy Refund object
+		def process_refund(refund)
+			raise NotImplementedError
+		end
```

Comment:
Do we want to keep that?

Reply:
Yes, the idea is that you need to override this in any subtransactable, like OfflineTransaction or StripeTransaction. That said, I should add a comment there to explain that.

Reply:
I've added an explanation to the error and documentation now